src/FeedIo/Formatter/XmlFormatter.php

```diff
@@ -140,10 +140,23 @@ public function toDom(FeedInterface $feed) : \DOMDocument
 
         $this->setHeaders($document, $feed);
         $this->setItems($document, $feed);
+        $this->setNS($document, $feed);
 
         return $document;
     }
 
+    public function setNS(\DOMDocument $document, FeedInterface $feed)
+    {
+        $firstChild = $document->firstChild;
+        foreach ($feed->getNS() as $ns) {
```

Comment:
If you set the namespace as the offset name as mentioned in the other comment, you will be able to write something like : 

```
foreach($feed->getNS() as $namespace => $dtd) {
```

which helps understand what the code does.